x/interchainquery/keeper/msg_server.go

```diff
@@ -57,40 +57,28 @@ func (k Keeper) VerifyKeyProof(ctx sdk.Context, msg *types.MsgSubmitQueryRespons
 	if !found {
 		return errorsmod.Wrapf(types.ErrInvalidICQProof, "ConnectionId %s does not exist", query.ConnectionId)
 	}
-
+	consensusState, found := k.IBCKeeper.ClientKeeper.GetClientConsensusState(ctx, connection.ClientId, height)
```

Comment:
I'm not very well versed on the internals of the proof itself, but the high level flow is:
* Look up connection
* Get client state and consensus state from connection
* Cast each to their respective tendermint types (since the type returned from the connection is a generic interface)
* Grab the Root from the consensus state, and the proof from the client state and pass them to the VerifyMembership functions

The only real change to these functions introduced by SDK 47, to my knowledge, is that previously the consensusState was a fixed type (instead of a generic interface), but has since been changed. So the casting to tendermint is a new required change. Additionally, the GetRoot function was removed and is now just a Root attribute.

The original PR that notional opened, generalized the consensus stae/client state casting in a switch statement so that it would work with types beyond tendermint. However, tendermint was still the only case implemented and we only support tendermint atm, so imo, the switch statement just made it a bit clunkier / harder to follow. This PR just removes the switch statement and hard codes it to tendermint, as well as fixes a bug where the `clientStateProof` was still uninitialized when it was passed into the VerifyMembership function (i.e. [this](https://github.com/Stride-Labs/stride/pull/781/files#diff-0c114c8ca86f30203e15e6b85a11b0b67e35bfa6d32072cedf664d69944da1d6L71) variable was never set in the switch statement before it was passed to VerifyMembership)

Reply:
That makes sense; thanks for elaborating. Let's push the review to the other PR, hard to disentangle Notional's changes being reverted vs the actual changes.